waku/v2/protocol/subscription/subscription_details.go

```diff
@@ -0,0 +1,122 @@
+package subscription
+
+import (
+	"encoding/json"
+	"sync"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/waku-org/go-waku/waku/v2/protocol"
+)
+
+// Map of SubscriptionDetails.ID to subscriptions
+type SubscriptionSet map[string]*SubscriptionDetails
+
+type PeerSubscription struct {
+	PeerID             peer.ID
+	SubsPerPubsubTopic map[string]SubscriptionSet
+}
+
+type PeerContentFilter struct {
+	PeerID        peer.ID  `json:"peerID"`
+	PubsubTopic   string   `json:"pubsubTopics"`
+	ContentTopics []string `json:"contentTopics"`
+}
+
+type SubscriptionDetails struct {
+	sync.RWMutex
+
+	ID     string `json:"subscriptionID"`
+	mapRef *SubscriptionsMap
+	Closed bool `json:"-"`
+	once   sync.Once
+
+	PeerID        peer.ID                 `json:"peerID"`
+	ContentFilter protocol.ContentFilter  `json:"contentFilters"`
+	C             chan *protocol.Envelope `json:"-"`
+}
+
+func (s *SubscriptionDetails) Add(contentTopics ...string) {
+	s.Lock()
+	defer s.Unlock()
+
+	for _, ct := range contentTopics {
+		if _, ok := s.ContentFilter.ContentTopics[ct]; !ok {
+			s.ContentFilter.ContentTopics[ct] = struct{}{}
+			// Increase the number of subscriptions for this (pubsubTopic, contentTopic) pair
+			s.mapRef.Lock()
+			s.mapRef.increaseSubFor(s.ContentFilter.PubsubTopic, ct)
+			s.mapRef.Unlock()
```

Comment:
I wonder if we shouldn't instead add a public `IncreaseSubFor` in `SubscriptionMap` that has this definition:
```go
func (m *SubscriptionsMap) IncreaseSubFor(pubsubTopic, contentTopic string) {
    m.Lock()
    defer m.Unlock()
    m.increaseSubFor(pubsubTopic, ct)
} 
```
That way, `SubscriptionDetails` does not need to know that `SubscriptionMap` has a mutex, but would simply call the new function. The same applies for `decreaseSubFor`